appjson/healthcheck.go

```diff
@@ -565,11 +570,14 @@ func (h Healthcheck) listeningCheck(container types.ContainerJSON) error {
 		addresses[parts[3]] = true
 	}
 
-	validAddresses := map[string]bool{
-		"0.0.0.0": true,
-		":::":     true,
+	if err := h.validateAddresses(addresses); err != nil {
+		return err
 	}
 
+	return nil
+}
+
+func (h Healthcheck) validateAddresses(addresses map[string]bool) error {
```

Comment:
thought: Extracting to logic out into `validateAddresses()` was key to be able to test as I tried writing a test for `listeningCheck()` but there was a _lot_ of setup required.